Core/Src/main.c

```diff
@@ -143,7 +172,34 @@ void SystemClock_Config(void)
 }
 
 /* USER CODE BEGIN 4 */
+uint16_t ADC_handle(void){
+
+	//Pull Chip select pin low to start connection
+	HAL_GPIO_WritePin(GPIOB, CS_PIN, 0);
+	uint8_t startByte = 0x01;
+
+	//send start bit
+	uint8_t garbage;
+	HAL_SPI_TransmitReceive(&hspi1, &startByte, &garbage, 1, 100);
+
+
+	uint8_t send[2], receive[2];
+	send[0] = (1<<7);
 
+	//Set channel to 0 and receive result of conversion
+	HAL_SPI_TransmitReceive(&hspi1, send, receive, 2, 100);
```

Comment:
I think I see what you are trying to do here. Althouh this may work, the function allows you to do all of this in one call.

So it looks like you know you have to send and recieve 3 bytes total, why not just make an array of size 3 for the tx and rx and do the transaction in one function call?